core/vm/gas_table.go

```diff
@@ -278,24 +278,59 @@ var (
 	gasMLoad   = pureMemoryGascost
 	gasMStore8 = pureMemoryGascost
 	gasMStore  = pureMemoryGascost
-	gasCreate  = pureMemoryGascost
 )
 
+func gasCreate(evm *EVM, contract *Contract, stack *Stack, mem *Memory, memorySize uint64) (uint64, error) {
+	gas, err := memoryGasCost(mem, memorySize)
+	if err != nil {
+		return 0, err
+	}
+	len, overflow := stack.Back(2).Uint64WithOverflow()
+	if overflow {
+		return 0, ErrGasUintOverflow
+	}
+
+	if hasEIP3860(&evm.Config) {
+		lenWords := toWordSize(len)
+		initCodeWordGas, overflow := math.SafeMul(lenWords, params.InitCodeWordGas)
+		if overflow {
+			return 0, ErrGasUintOverflow
+		}
+		if gas, overflow = math.SafeAdd(gas, initCodeWordGas); overflow {
+			return 0, ErrGasUintOverflow
+		}
+	}
+	return gas, nil
+}
```

Comment:
Another way to achieve this, is to define `gasCreateEip3860`. And in the `core/vm/eips.go`, you define a "new" create-operation which uses the new gas calculator. 

That approach is IMO better, because it doesn't add these checks to various places and makes things more complicated, but rather keeps hardfork-changes separate. The downside is that we get more different methods, and a bit more copy-paste functions, but it's still preferrable in my opinion,